app/gui/src/profile_workflow.rs

```diff
@@ -0,0 +1,118 @@
+//! Defines profilable workflows, and an entry point that runs a specified workflow.
+
+use crate::integration_test::prelude::*;
+use wasm_bindgen::prelude::*;
+
+use enso_debug_api as debug_api;
+use enso_web as web;
+
+
+
+// ====================
+// === ReflectMatch ===
+// ====================
+
+/// Match a value against a set of candidates; if no match is found, the list of candidates is
+/// available.
+macro_rules! reflect_match {
+    (@acc ($dispatch:ident, $value:expr, $candidates:ident, {
+        _ => $fallback:expr $(,)?
+    }) -> {$( $branches:tt )*}) => {
+        let mut $dispatch = ReflectMatch::new($value);
+        match () {
+            $( $branches )*
+            _ => {
+                let $candidates = $dispatch.candidates;
+                $fallback
+            }
+        }
+    };
+    (@acc ($dispatch:ident, $value:expr, $candidates:ident, {
+        $candidate:literal => $branch:expr,
+        $( $rest:tt )*
+    }) -> {$( $branches:tt )*}) => {
+        reflect_match!(@acc ($dispatch, $value, $candidates, { $( $rest )* }) -> {
+            $( $branches )*
+            _ if $dispatch.matches($candidate) => $branch,
+        })
+    };
+    ($candidates:ident, match $value:tt { $( $branches:tt )* }) => {
+        reflect_match!(@acc (dispatch, $value, $candidates, { $( $branches )* }) -> {})
+    };
+}
+
+
+// === ReflectMatch ===
+
+/// Used to match a value against a set of candidates, while keeping track of the candidates.
+struct ReflectMatch<T, U> {
+    value:      T,
+    candidates: Vec<U>,
+}
+
+impl<T, U> ReflectMatch<T, U> {
+    /// Create a new dispatcher, for a given value.
+    fn new(value: T) -> Self {
+        let candidates = Default::default();
+        Self { value, candidates }
+    }
+
+    /// Test the value against a candidate. Return whether it's a match.
+    fn matches(&mut self, key: U) -> bool
+    where T: PartialEq<U> {
+        let matches = self.value == key;
+        self.candidates.push(key);
+        matches
+    }
+}
+
+
+
+// ===================
+// === Entry point ===
+// ===================
+
+#[wasm_bindgen]
+#[allow(dead_code)]
+pub async fn entry_point_profile() {
+    web::forward_panic_hook_to_console();
+    ensogl_text_msdf_sys::initialized().await;
+
+    // Run selected workflow.
+    let need_workflow = "`profile` entry point requires --workflow argument. \
+    Try --workflow=help to see a list of options.";
+    let selected = enso_config::ARGS.workflow.as_ref().expect(need_workflow);
+    reflect_match!(options, match selected {
+        "add_node" => profile_add_node().await,
+        "new_project" => profile_new_project().await,
+        _ => panic!("Unknown workflow: {selected}. Must be one of: {options:?}."),
+    });
```

Comment:
1. I would change the syntax to `reflect_match!(options = match selected ...` to show to the reader that `options` is generated from the call, what do you think about it?
2. This is an extremely cool structure, it might be helpful in other places, maybe we should refactor it somewhere?
3. The docs for the macro should provide an example with explanation how it works. I needed to spend some time to understand what it really does - the docs are very generic and does not explain the usage.

Reply:
1.I think the assignment syntax could be confusing if the `match` expression is used as a value; also, typically the binding of an assignment is in scope only after the assignment, but in this case the binding would only be in scope *during* the assignment. What about this:
```rust
reflect_match!(as options, match selected {
```
2 & 3: Documented and moved to `enso_prelude`.